### PR TITLE
S0067 masked modulus10 13

### PR DESCRIPTION
**Pull request by KnowledgeForwardSolutions**
This pull request introduces support for masked check digit validation to the `Modulus10_13Algorithm`, adds a new `MaskedAlgorithms` singleton class, and provides comprehensive tests for masked algorithms. It also includes minor improvements to documentation and utility methods. The most important changes are grouped below by theme.

### Masked Algorithm Support

* [`CheckDigits.Net/GeneralAlgorithms/Modulus10_13Algorithm.cs`](diffhunk://#diff-223d7285dd375cff0f5e90bdd3604020d570e971ab6da8774fb8e358f668b68fL25-R25): Added implementation of the `IMaskedCheckDigitAlgorithm` interface to `Modulus10_13Algorithm`, including a new overload of `Validate` that accepts an `ICheckDigitMask` and processes masked validation logic. [[1]](diffhunk://#diff-223d7285dd375cff0f5e90bdd3604020d570e971ab6da8774fb8e358f668b68fL25-R25) [[2]](diffhunk://#diff-223d7285dd375cff0f5e90bdd3604020d570e971ab6da8774fb8e358f668b68fR85-R120)
* [`CheckDigits.Net/MaskedAlgorithms.cs`](diffhunk://#diff-ff964210b26c6fa4f14ae25a56e4cd7a73e5a962a77999317c4b0794acab6711R1-R27): Introduced the `MaskedAlgorithms` static class, providing lazy singleton instances of masked algorithms (`LuhnAlgorithm` and `Modulus10_13Algorithm`).

### Testing Enhancements

* [`CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_13AlgorithmTests.cs`](diffhunk://#diff-aaecbb241eb19364911d8f5b03e874b35df9259549de99b7051d496dfc28a14fR189-R278): Added extensive tests for masked validation, including edge cases, correct weighting, valid/invalid check digits, and error scenarios.
* [`CheckDigits.Net.Tests.Unit/MaskedAlgorithmsTests.cs`](diffhunk://#diff-4dd096f96ce85b178672dab194e55dba9b01c61c59201fc2662072de29a1837aR1-R34): Added new unit tests for the `MaskedAlgorithms` singleton class, confirming correct instantiation and types.

### Codebase Cleanup

* [`CheckDigits.Net/IEmbeddedCheckDigitAlgorithm.cs`](diffhunk://#diff-0042328a09d47942383d0d03d6996f9f2c4786c7071194ba88c1c2ccd5463915L1-L40): Removed the unused `IEmbeddedCheckDigitAlgorithm` interface.
* [`CheckDigits.Net/Utility/ExtensionMethods.cs`](diffhunk://#diff-4f0c25098594b8d699cb74b667c4c8228d2abd7ce3ef62d7349119fbf2b2052cR35): Added aggressive inlining to the `ToIntegerDigit` extension method for performance.

### Documentation Updates

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L126-R128): Updated documentation to clarify usage of masked algorithms, correct code formatting, and reflect the new support for `Modulus10_13Algorithm` in masked validation. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L126-R128) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L220-R220) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L251-R256) [[4]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L307-R315) [[5]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L325-R352) [[6]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L395-R396) [[7]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L416-R417) [[8]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L447-R448)

### Minor Improvements

* [`CheckDigits.Net/Utility/ExtensionMethods.cs`](diffhunk://#diff-4f0c25098594b8d699cb74b667c4c8228d2abd7ce3ef62d7349119fbf2b2052cL1-R3): Added `using System.Runtime.CompilerServices` for method inlining.

These changes collectively improve the flexibility and robustness of check digit validation, especially for cases where masking is required.

### CODE REVIEW

File: CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_13AlgorithmTests.cs [hunk 1]
```diff
@@ -2,186 +2,279 @@
...
+   [InlineData("03600029145", '2')]          // Worked UPC-A example from Wikipedia (https://en.wikipedia.org/wiki/Universal_Product_Code#Check_digit_calculation)
+   [InlineData("42526", '1')]                // UPC-E example
+   [InlineData("400638133393", '1')]         // Worked EAN-13 example from Wikipedia (https://en.wikipedia.org/wiki/International_Article_Number)
+   [InlineData("7351353", '7')]              // Worked EAN-8 example from Wikipedia
+   [InlineData("978050051695", '9')]         // ISBN-13, Islamic Geometric Design, Eric Broug
+   [InlineData("01234567800004567", '8')]    // Example SSCC number
+   public void Modulus10_13Algorithm_TryCalculateCheckDigit_ShouldCalculateExpectedCheckDigit(
+      String value,
+      Char expectedCheckDigit)
+{
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply CheckDigits.Net.Tests.Unit/GeneralAlgorithms/Modulus10_13AlgorithmTests.cs]: will update
